### DISCUSSION TITLE
Cascading Filters

### DISCUSSION THREAD

**Discussion by LifeofGlen** [Budibase/budibase#14373] [Help] [answered]
Hi all,

I was going through this tutorial in the docs section:
https://docs.budibase.com/docs/cascading-dropdown-filters

and i noticed that the demo does not work as described. I have also tested in my own app and can see that the final step 16 & 17 do not work:

"16. Finally we want to clear the selected cities when the country changes. Click on the Country Options Picker, and click the button beside the On change setting
17. Add the Update Field Value action:
-       Select the Form
-       Set Type to 'Reset to default value'
-       Select the 'Cities' Field from the dropdown
-       Be sure to Save"

When you select a new country, the old cities remain in the drop down selection. I am struggling to make this work on my side. Any assistance would be appreciated.

screenshots:

![image](https://github.com/user-attachments/assets/9a27a8ef-8b9a-424c-9bb5-4875baedb809)

**Comment by mikesealey** [comment, marked as answer]
Hey @LifeofGlen I'm sorry to hear you're having this issue. I tried working through the tutorial myself and found the same issue. It seems to have been raised in the past in this issue: https://github.com/Budibase/budibase/issues/13893 but possibly due to the issue being closed and reopened again it may have gone unnoticed. I'll raise it with the engineering team.

For the time being, you can work around this issue by using an extra Data Provider. Add a new Data Provider as a sibling of the options picker, and move the relationship picker into it. 
![image](https://github.com/user-attachments/assets/db2c6d76-b2b3-471f-a1cf-74388fc7fc48)

Set the Data Provider to look at the Cities table, and filter on the City's country_id being equal to the form field Country.
![image](https://github.com/user-attachments/assets/59309d84-8936-412c-be20-e0f8a0692056)

It has the same effect, but it a little less streamlined. Hopefully that gets you unstuck, but if there's anything else you need just ask.
